apiserver/facades/controller/caasapplicationprovisioner/state.go

```diff
@@ -111,6 +113,15 @@ func (s stateShim) Resources() Resources {
 	return s.State.Resources()
 }
 
+func (s stateShim) Unit(unitTag string) (Unit, error) {
+	unit, err := s.State.Unit(unitTag)
+	if err != nil {
+		return nil, err
+	}
+
+	return unit, nil
```

Comment:
Seems like we should just `return s.State.Unit(unitTag)`. Otherwise, maybe we should trace/annotate the error.